pay-api/src/pay_api/services/statement_settings.py

```diff
@@ -0,0 +1,185 @@
+        self.__dao = None
+        self._id: int = None
+        self._frequency = None
+        self.payment_account_id = None
+        self._from_date = None
+        self._to_date = None
+
+    @property
+    def _dao(self):
+        if not self.__dao:
+            self.__dao = StatementSettingsModel()
+        return self.__dao
+
+    @_dao.setter
+    def _dao(self, value):
+        self.__dao = value
+        self.id: int = self._dao.id
+        self.frequency: str = self._dao.frequency
+        self.payment_account_id: int = self._dao.payment_account_id
+        self.from_date: datetime = self._dao.from_date
+        self.to_date: datetime = self._dao.to_date
+
+    @property
+    def frequency(self):
+        """Return the for the statement settings."""
+        return self._frequency
+
+    @frequency.setter
+    def frequency(self, value: int):
+        """Set the frequency for the statement settings."""
+        self._frequency = value
+        self._dao.frequency = value
+
+    @property
+    def id(self):
+        """Return the _id."""
+        return self._id
+
+    @id.setter
+    def id(self, value: int):
+        """Set the id."""
+        self._id = value
+        self._dao.id = value
+
+    @property
+    def from_date(self):
+        """Return the from_date of the statement setting."""
+        return self._from_date
+
+    @from_date.setter
+    def from_date(self, value: date):
+        """Set the from for the statement setting."""
+        self._from_date = value
+        self._dao.from_date = value
+
+    @property
+    def payment_account_id(self):
+        """Return the account_id."""
+        return self._payment_account_id
+
+    @payment_account_id.setter
+    def payment_account_id(self, value: str):
+        """Set the account_id."""
+        self._payment_account_id = value
+        self._dao.payment_account_id = value
+
+    @property
+    def to_date(self):
+        """Return the to_date of the statement setting."""
+        return self._to_date
+
+    @to_date.setter
+    def to_date(self, value: date):
+        """Set the to_date for the statement setting."""
+        self._to_date = value
+        self._dao.to_date = value
+
+    def asdict(self):
+        """Return the invoice as a python dict."""
+        statements_settings_schema = StatementSettingsModelSchema()
+        d = statements_settings_schema.dump(self._dao)
+        return d
+
+    @staticmethod
+    def find_by_account_id(account_id: str):
+        """Find statements by account id."""
+        current_app.logger.debug(f'<find_by_account_id {account_id}')
+        statements_settings = StatementSettingsModel.find_latest_settings(account_id)
+        if statements_settings is None:
+            # if empty , default to weekly
+            statements_settings = StatementSettingsModel(frequency=StatementFrequency.WEEKLY.value,
+                                                         payment_account_id=account_id)
```

Comment:
Also this payment_account_id population is not correct. Look up the PaymentAccount with auth_account_id and then add payment_account.id as payment_account_id